Automattic/wp-calypso
# Update atomic hosting sftp card to use new data-layer state

#### Changes proposed in this Pull Request

* Refactor the sftp card component to use data layer state from #37430
* Add API error handling to fix #37256

#### Testing instructions

Because the change affects all the sftp user flows (view, add, reset password) these all need to be retested.

* Apply this patch to you Calypso dev environment
* Select an Atomic site that does not have an SFTP user created
* View the /hosting admin section for site and check that the `Enable SFTP` button displays
* Click the button and check that user is created and username and password display
* Navigate away from the component and then back to it and check that the username is still displayed but password is empty
* Click the `Reset Password` button and check that you get a new password displayed
* Navigate away and back and check that the password is empty again
* Simulate a network error when resetting password and check that an error notification displays as below

<img width="983" alt="Screen Shot 2019-11-08 at 11 28 45 AM" src="https://user-images.githubusercontent.com/3629020/68433844-6bd9d800-021c-11ea-8325-8b8163419278.png">
